modules/caddyhttp/reverseproxy/caddyfile.go

```diff
@@ -719,6 +719,14 @@ func (h *HTTPTransport) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 					return d.ArgErr()
 				}
 
+			case "compression":
+				if d.NextArg() {
+					if d.Val() == "off" {
+						var disable bool
+						h.Compression = &disable
```

Comment:
Why use this approach to setting it with a reference? Why not just set it to false directly?

```suggestion
						h.Compression = false
```

Reply:
Also could you please add an adapt test for this? See `caddytest/integration/caddyfile_adapt/*`

Reply:
> Why use this approach to setting it with a reference? Why not just set it to false directly?

This approach is imitating a legacy practice in setting another field:
https://github.com/caddyserver/caddy/blob/19cc2bd3c3e6b551d47eca7a2efdee95eec2a86f/modules/caddyhttp/reverseproxy/caddyfile.go#L684-L695

Here `h.Compression` is of type `*nool`, so cannot just bind false to it...

Reply:
> Also could you please add an adapt test for this? See `caddytest/integration/caddyfile_adapt/*`

I've added an adapt test for a typical h2c reverse_proxy setup that conforms to the JSON config in https://github.com/caddyserver/caddy/issues/3556#issuecomment-655274156.

Reply:
Thanks!

Reply:
The reason the boolean is a pointer is because we need a way to distinguish between the user explicitly setting it to false and it being unset, because the default is true.